biztown/src/components/FormInputLayout.js

```diff
@@ -1,260 +1,428 @@
-            ...this.state.freeResponseData.checkSlip
-        }
-        console.log("completedForm",completedForm)
+    if (currentStep === 2 && this.validatePage(multipleChoiceDataObj)) {
+      incrementStepAndUpdateState();
+      return;
+    }
 
-        var submitPath = this.props.posttest ? "postassessment/" : "preassessment/";
-        DataManager.post(submitPath, completedForm).then(() => {
-          <Redirect to="/completionPage" />
-        })
+    if (currentStep === 3) {
+      incrementStepAndUpdateState();
+      return;
     }
-    
-    _next = e => {
-        // this.handleSubmit(e)
-        let currentStep = this.state.currentStep
-        currentStep = currentStep >= 5 ? 6 : currentStep + 1
-        this.setState({
-            currentStep: currentStep
-        })
-        e.preventDefault();
+
+    if (currentStep === 4 && this.validatePage(aboutMeObj)) {
+      incrementStepAndUpdateState();
+      return;
     }
-      
-    _prev = () => {
-        let currentStep = this.state.currentStep
-        currentStep = currentStep <= 1? 1: currentStep - 1
-        this.setState({
-            currentStep: currentStep
-        })
+
+    if (
+      currentStep === 5 &&
+      this.props.postTest &&
+      this.validatePage(aboutMyFutureObj)
+    ) {
+      incrementStepAndUpdateState();
+      return;
     }
+  };
 
-    /*
-    * the functions for our button
-    */
-    previousButton() {
-        let currentStep = this.state.currentStep;
-        if(currentStep !==1){
-            return (
-                <button 
-                    className="btn btn-secondary navigation-btn" 
-                    type="button" onClick={this._prev}>
-                Previous
-                </button>
-            )
-        }
-        return null;
+  _prev = () => {
+    let currentStep = this.state.currentStep;
+    currentStep = currentStep <= 1 ? 1 : currentStep - 1;
+    this.setState({
+      currentStep: currentStep,
+    });
+  };
+
+  /*
+   * the functions for our button
+   */
+  previousButton() {
+    let currentStep = this.state.currentStep;
+    if (currentStep !== 1) {
+      return (
+        <button
+          className="btn btn-secondary navigation-btn"
+          type="button"
+          onClick={this._prev}
+        >
+          Previous
+        </button>
+      );
     }
+    return null;
+  }
 
-    nextButton(){
-        let currentStep = this.state.currentStep;
-        if(currentStep < 6){
-            return (
-                <button 
-                    className="btn btn-primary float-right navigation-btn" 
-                    type="button" onClick={this._next}>
-                    Next
-                </button>        
-            )
-        }
-        return null;
+  nextButton() {
+    let currentStep = this.state.currentStep;
+    if (currentStep === 5 && this.props.preTest) {
+      return (
+        <button className="btn btn-success btn-block float-right navigation-btn">
```

Comment:
It is a little strange having the submit button both contained within the `nextButton` and the `submitButton`. This can be condensed into having it just in the `submitButton`,

Reply:
Done. Also, should I be the one 'resolving' the conversation or am I meant to implement the changes, push them, and after they're reviewed the reviewer resolves them?